### ISSUE TITLE
generic writer performance ticket

### ISSUE THREAD

**Issue by caolanm** [CollaboraOnline/online#8051]
placeholder to track weekly writer profiles

**Comment by caolanm** [comment]
@mmeeks @grandinj Seeing as a generally get a writer profile once a week, I'll stash it here regularly. 

This weeks writer profile:

![perf-677376](https://github.com/CollaboraOnline/online/assets/833656/a4e87e6d-74dc-482b-aacd-855eff1ce1a8)

**Comment by mmeeks** [comment]
Looks to me like we've got some galloping invalidation problem that is causing lots of tiles to be rendered un-necessarily. I would expect to see the event handling, spell-checking, editing of the document etc. to be a much larger proportion of the cost cf. a more normal profile like this from mid July:
![perf-23 05-writer](https://github.com/CollaboraOnline/online/assets/122848/61cb4dbc-fde9-4f40-863a-a3c3a6a9b578)

**Comment by caolanm** [comment]
![perf-128281](https://github.com/CollaboraOnline/online/assets/833656/b9d2e818-1db0-410c-8e4e-693421ab184c)

About 20 minutes in writer. AsynchronousCall::Sync call seems to have emerged as a big cost. @vmiklos maybe https://gerrit.libreoffice.org/c/core/+/161952 is a problem there after all

**Comment by eszkadev** [comment]
This probably is regression from my recent commit, have fix for it: https://github.com/CollaboraOnline/online/pull/8092

**Comment by vmiklos** [comment]
I hope that the above helps; setting the language/locale is meant to be cheap, so I would be a bit puzzled in case doing that would have performance impact. But I haven't measured.

**Comment by caolanm** [comment]
![perf-179814](https://github.com/CollaboraOnline/online/assets/833656/96b751cb-2703-40a5-8228-4566dee973ed)

This weeks writer profile

**Comment by caolanm** [comment]
![perf-294309](https://github.com/CollaboraOnline/online/assets/833656/5339f5ba-32fd-4331-a143-bc3d727898ed)

todays ad-hoc test, most longer document than usual (86 pages) heavy with images

**Comment by mmeeks** [comment]
Hmm; I'm surprised to see the DrawBitmapEx doing that slowish clip/composite path; is there some rotation or other going on there? trace is a bit odd too, might be nice to have a sample doc with the image in it attached; I wonder if Armin has any ideas. Still lots of time in the PaintBaBo and PaintBorderAndShadow which seems to be doing something un-necessary with an alpha bitmap too =)

**Comment by caolanm** [comment]
Today's uneventful writer session

![perf-527342](https://github.com/CollaboraOnline/online/assets/833656/f8caf232-aa03-43b1-948f-1aacb2142f65)

**Comment by caolanm** [comment]
![perf-132657](https://github.com/CollaboraOnline/online/assets/833656/41d4ea5d-46c2-499c-8a17-b0685924c755)

Todays writer flamegraph

**Comment by mmeeks** [comment]
~18% of the time in GetWaterMark is indeed interesting; and it seems there is far too much invalidation & re-layout going on playing with the document.

**Comment by caolanm** [comment]
This weeks regular writer profile

![perf-14543](https://github.com/CollaboraOnline/online/assets/833656/2b307bed-fb5a-46cf-badb-31d91a66621f)

**Comment by caolanm** [comment]
I'll put writer 24.04 flamegraphs at: https://github.com/CollaboraOnline/online/issues/8571

**Comment by caolanm** [comment]
Todays (23.05) writer profile

![perf-81670](https://github.com/CollaboraOnline/online/assets/833656/15ebee47-c4b8-475f-912a-d1927e3722ac)

**Comment by mmeeks** [comment]
That's quite encouragingly sensible; I guess - at least this looks plausibly like what I would expect us to do; clearly shrinking each piece is good but overall it looks balanced enough ? =)

**Comment by caolanm** [comment]
Nothing terrible, but it was a simple session. Maybe ApplyViewOptions is a little suspicious, and presumably the xor stuff under that is from a blinking cursor which is presumably the kind of thing that doesn't need to happen in kit-mode.

**Comment by caolanm** [comment]
![perf-579991](https://github.com/CollaboraOnline/online/assets/833656/855e7b30-7f0c-440a-9a1e-22003b6ac8ad)

fairly unremarkable 23.05 writer session perf-wise I think

**Comment by timur-g** [comment]
I assume this may be closed.

**Comment by jazevedo-coll** [comment]
Yes, we have other such trackers for 25.04, so this can be closed.
